### ISSUE TITLE
prometheus does not start with error "Unable to create mmap-ed active query log"

### ISSUE THREAD

**Issue by ricsanfre** [ricsanfre/pi-cluster#9]
Trying to deploy helm kube-prometheus-stack providing values.yml where volume claim templates are configured to use longhorn, prometheus pod did not start.

Logs show the following error:

* "Error opening query log file" file=/prometheus/queries.active err="open /prometheus/queries.active: permission denied"
panic: Unable to create mmap-ed active query log*

Helm values overriden
`values.yml`: 

```yml
alertmanager:
        alertmanagerSpec:
          storage:
            volumeClaimTemplate:
              spec:
                storageClassName: longhorn
                accessModes: ["ReadWriteOnce"]
                resources:
                  requests:
                    storage: 5Gi
      prometheus:
        prometheusSpec:
          storageSpec:
            volumeClaimTemplate:
              spec:
                storageClassName: longhorn
                accessModes: ["ReadWriteOnce"]
                resources:
                  requests:
                    storage: 5Gi

```

Pod logs:

```
kubectl logs prometheus-kube-prometheus-stack-prometheus-0 -n k3s-monitoring
level=info ts=2021-11-02T16:20:48.163Z caller=main.go:443 msg="Starting Prometheus" version="(version=2.28.1, branch=HEAD, revision=b0944590a1c9a6b35dc5a696869f75f422b107a1)"
level=info ts=2021-11-02T16:20:48.163Z caller=main.go:448 build_context="(go=go1.16.5, user=root@fa052cd511a1, date=20210701-15:08:46)"
level=info ts=2021-11-02T16:20:48.163Z caller=main.go:449 host_details="(Linux 5.4.0-1045-raspi #49-Ubuntu SMP PREEMPT Wed Sep 29 17:49:16 UTC 2021 aarch64 prometheus-kube-prometheus-stack-prometheus-0 (none))"
level=info ts=2021-11-02T16:20:48.163Z caller=main.go:450 fd_limits="(soft=1048576, hard=1048576)"
level=info ts=2021-11-02T16:20:48.163Z caller=main.go:451 vm_limits="(soft=unlimited, hard=unlimited)"
level=error ts=2021-11-02T16:20:48.164Z caller=query_logger.go:87 component=activeQueryTracker msg="Error opening query log file" file=/prometheus/queries.active err="open /prometheus/queries.active: permission denied"
panic: Unable to create mmap-ed active query log

goroutine 1 [running]:
github.com/prometheus/prometheus/promql.NewActiveQueryTracker(0xfffff37ab1ba, 0xb, 0x14, 0x2ee5e18, 0x40001feff0, 0x2ee5e18)
        /app/promql/query_logger.go:117 +0x3a4
main.main()
        /app/cmd/prometheus/main.go:479 +0x4de4
```

**Comment by ricsanfre** [comment]
Same issue as https://github.com/aws/eks-charts/issues/21. 

It seems to be a problem with the  permissions associated to the PVC since prometheus pod is running as non-root user.
By default ownership is root:root 755. This means that volumes used this way require additional configuration steps in order to be usable by non-root containers, using a initContainer changing privileges of the pod.

As an alternative security context for prometheus pods can be changed as stated in kubernetes [documentation](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod)

**Comment by ricsanfre** [comment]
Longhorn v1.2.1 seems to solve `fsgroup` issue and the workaround it is not needed anymore.

**Comment by ricsanfre** [comment]
Issue solved using Longhorn v1.2.2.
